packages/contracts/contracts/plugin-trait.clar

```diff
@@ -0,0 +1,6 @@
+(define-trait plugin-trait 
+  (
+    (get-name () (response (string-ascii 50) bool))
+    (get-version () (response uint bool))
```

Comment:
Clarity noob questions:
1. I'm used to seeing `uint` where `bool` is to specify a returned error code. What is the significance of using `bool` here? Does this mean we are returning `true`/`false` instead of returning an error and that's why we aren't using `?` after the function name? i.e this function will never return an error?
2. Is there no way to specify a trait function should be `define-read-only`?

Could we name this trait something more descriptive like `plugin-info` instead of `plugin-trait`?

:)

Reply:
**Ad 1)** You are 100% correct. It is a mistake on my end. I misunderstood how functions should be defined in trait definition and thought that second value of response should be `bool` not `uint`.

**Ad 2)** Not at this moment. There is a proposal for that: https://github.com/blockstack/stacks-blockchain/issues/1981

We can name it `plugin-info` however I believe that we'll have more functions that all plugins must implement, so I picked `plugin-trait`.
```suggestion
    (get-name () (response (string-ascii 50) uint))
    (get-version () (response uint uint))
```